packages/next/next-server/server/render.tsx

```diff
@@ -546,6 +546,18 @@ export async function renderToHTML(
     props.pageProps = data.props
     ;(renderOpts as any).pageData = props
   }
+
+  if (
+    !isSpr && // we only show this error for legacy pages
+    !unstable_getServerProps &&
+    process.env.NODE_ENV !== 'production' &&
+    Object.keys(props?.pageProps || {}).includes('url')
+  ) {
+    throw new Error(
```

Comment:
This shouldn't be added in this PR imo. It's a breaking change for anyone using getInitialProps and there's really nothing to gain from throwing here.

Reply:
Updated